internal/bminventory/inventory_v2_handlers.go

```diff
@@ -778,20 +778,51 @@ func (b *bareMetalInventory) GetInfraEnvDownloadURL(ctx context.Context, params
 	return installer.NewGetInfraEnvDownloadURLOK().WithPayload(&models.PresignedURL{URL: &newURL, ExpiresAt: *expiresAt})
 }
 
-func (b *bareMetalInventory) generateImageDownloadURL(ctx context.Context, infraEnvID, imageType, version, arch, imageTokenKey string) (string, *strfmt.DateTime, error) {
-	urlString, err := imageservice.ImageURL(b.ImageServiceBaseURL, infraEnvID, version, arch, imageType)
+func (b *bareMetalInventory) generateShortImageDownloadURL(infraEnvID, imageType, version, arch, imageTokenKey string) (string, *strfmt.DateTime, error) {
+	switch b.authHandler.AuthType() {
+	case auth.TypeLocal:
+		return b.generateShortImageDownloadURLByAPIKey(infraEnvID, imageType, version, arch)
+	case auth.TypeRHSSO:
+		return b.generateShortImageDownloadURLByToken(infraEnvID, imageType, version, arch, imageTokenKey)
+	case auth.TypeNone:
+		return b.generateShortImageDownloadURLByID(infraEnvID, imageType, version, arch)
+	}
+
+	return "", nil, errors.Errorf("Unknown auth type %s", b.authHandler.AuthType())
+}
+
+func (b *bareMetalInventory) generateShortImageDownloadURLByAPIKey(infraEnvID, imageType, version, arch string) (string, *strfmt.DateTime, error) {
+	var expiresAt strfmt.DateTime
+
+	token, err := gencrypto.LocalJWT(infraEnvID, gencrypto.InfraEnvKey)
 	if err != nil {
-		return "", nil, err
+		return "", nil, errors.Wrapf(err, "failed to generate token for infraEnv %s", infraEnvID)
 	}
-	urlString, err = b.signURL(ctx, infraEnvID, urlString, imageTokenKey)
+
+	shortURL, err := imageservice.ShortImageURL(b.ImageServiceBaseURL, imageservice.ByAPIKeyPath, token, version, arch, imageType)
+	return shortURL, &expiresAt, err
+}
+
+func (b *bareMetalInventory) generateShortImageDownloadURLByToken(infraEnvID, imageType, version, arch, imageTokenKey string) (string, *strfmt.DateTime, error) {
+	token, err := gencrypto.JWTForSymmetricKey([]byte(imageTokenKey), b.ImageExpirationTime, infraEnvID)
 	if err != nil {
-		return "", nil, err
+		return "", nil, errors.Wrap(err, "failed to sign image URL")
 	}
-	expiresAt, err := gencrypto.ParseExpirationFromURL(urlString)
+
+	exp, err := gencrypto.ParseExpirationFromToken(token)
 	if err != nil {
 		return "", nil, err
 	}
-	return urlString, expiresAt, nil
+
+	shortURL, err := imageservice.ShortImageURL(b.ImageServiceBaseURL, imageservice.ByTokenPath, token, version, arch, imageType)
+	return shortURL, exp, err
+}
+
+func (b *bareMetalInventory) generateShortImageDownloadURLByID(infraEnvID, imageType, version, arch string) (string, *strfmt.DateTime, error) {
+	var expiresAt strfmt.DateTime
```

Comment:
we don't need to set expiration date to some value?

Reply:
`nil` should be returned in case of `byid` and `byapikey`, will fix it

Reply:
actually no, before we were calling `ParseExpirationFromURL`: https://github.com/openshift/assisted-service/pull/5523/files#diff-9f6fae8a28b132e15c426658b6b5d41adf8a20006ddeb271a22fea270c573610R85-R96

which returns the default value of `strfmt.DateTime` when no `image _token` is passed.

Reply:
We have tests here:
https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory_test.go#L18547
https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory_test.go#L18579
https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory_test.go#L18601

Reply:
so for all cases we're always returning an expiration date of `0001-01-01T00:00:00.000Z`?

Reply:
no, in case of `image_token` we don't expect `0001-01-01T00:00:00.000Z`

Reply:
ack. thanks for the explanation!